react-native-aztec/android/src/main/java/org/wordpress/mobile/ReactNativeAztec/ReactAztecManager.java

```diff
@@ -498,22 +499,26 @@ public void onTextChanged(CharSequence s, int start, int before, int count) {
                 return;
             }
 
-            int currentEventCount = mEditText.incrementAndGetEventCounter();
-            // The event that contains the event counter and updates it must be sent first.
-            // TODO: t7936714 merge these events
-            mEventDispatcher.dispatchEvent(
-                    new ReactTextChangedEvent(
-                            mEditText.getId(),
-                            mEditText.toHtml(false),
-                            currentEventCount));
-
-            mEventDispatcher.dispatchEvent(
-                    new ReactTextInputEvent(
-                            mEditText.getId(),
-                            newText,
-                            oldText,
-                            start,
-                            start + before));
+            // if the "Enter" handling is underway, don't sent text change events. The ReactAztecEnterEvent will have
+            // the text (minus the Enter char itself).
+            if (mEditText.getText().getSpans(0, 0, EnterPressedUnderway.class).length == 0) {
```

Comment:
Nitpick: What do you think about moving `mEditText.getText().getSpans(0, 0, EnterPressedUnderway.class).length == 0` to a util method in `EnterPressedUnderway` or `AztecText`?

Reply:
Sounds good, will do 👍

Reply:
Done with 991f8a83.